include/hipSYCL/sycl/libkernel/sscp/builtins.hpp

```diff
@@ -550,14 +550,21 @@ HIPSYCL_BUILTIN T __hipsycl_fast_normalize(T a) noexcept {
 
 // ****************** relational functions ******************
 
-HIPSYCL_BUILTIN int __hipsycl_isnan(float x) {
-  return __hipsycl_sscp_isnan_f32(x);
-}
+#define HIPSYCL_DEFINE_SSCP_GENFLOAT_REL_BUILTIN(name)                         \
+  HIPSYCL_BUILTIN int __hipsycl_##name(float x) {                              \
+    return __hipsycl_sscp_##name##_f32(x);                                     \
+  }                                                                            \
+  HIPSYCL_BUILTIN int __hipsycl_##name(double x) {                             \
+    return __hipsycl_sscp_##name##_f64(x);                                     \
+  }
```

Comment:
Macros here might be similarly questionable as described in my comment further down :)

Since `__hipsycl_sscp_<name>` builtins return bool, maybe add a `static_cast` since we are returning int?

Reply:
I went in the other direction and instead changed the return types of the callees to `int` since that's what the NVVM, OCML and SPIR-V bitcode functions all give you.

Reply:
Ok.